### PR TITLE
[MIRROR] [MDB IGNORE] External Organ Rework: new bodypart_overlay system

### PR DESCRIPTION
**Pull request by TaleStationBot**
Original PR: https://github.com/tgstation/tgstation/pull/72734
-----
Bodypart overlays are now drawn by the new /datum/bodypart_overlay datum. 

External organs no longer draw anything and instead add a special /datum/bodypart_overlay/mutant to the bodypart, which draws everything

Makes it way easier to add custom overlays to limbs, since the whole system is now modularized and external organs are just one implementation of it

I haven't moved anything but external organs to this new system, I'll move eyes, bodymarkings, hair, lipstick etc to this later

New pipeline is as follows:
- External organ added to limb
- External organ adds /datum/bodypart_overlay/mutant to limb to bodypart_overlays
- Limb updates its icon, looks for all /datum/bodypart_overlay in bodypart_overlays
- Very cool new overlay on your limb!

closes #71820

:cl:
refactor: External organs have been near-completely refactored. 
admin: Admin-spawned external organs will load with a random icon and color
fix: fixes angel wings not working for non-humans (it was so fucking broken)
fix: fixes external organs being invisible if they werent initialized with a human
/:cl:

### Why this is good for the game
External organs are cool but are pretty limited in some ways. Making stuff like synthetic organs is kinda fucked. I tried and it was dogshit. Now you can just give an icon state and icon and you're good (using /datum/bodypart_accessory/simple)

Stuff like eyes, cat ears and hair seem like good choices for extorgans, but don't quite work for it because their icons work a lot differently. This solves for it completely since any organ (or object or whatever) can add it's own icon to a bodypart. 

Want to add an iron plate to someones head? Go ahead. Want a heart to stick out of someones chest? No problem.

### CODE REVIEW

File: code/modules/surgery/organs/external/tails.dm [hunk 1]
```diff
@@ -52,92 +41,115 @@
...
+	accessory.wagging = FALSE
+
+///Tail parent type (which is MONKEEEEEEEEEEE by default), with wagging functionality
+/datum/bodypart_overlay/mutant/tail
+	layers = EXTERNAL_FRONT|EXTERNAL_BEHIND
+	feature_key = "tail_monkey"
+	var/wagging = FALSE
+
+/datum/bodypart_overlay/mutant/tail/get_base_icon_state()
+	return (wagging ? "wagging_" : "") + sprite_datum.icon_state //add the wagging tag if we be wagging
```
Comment (Time-Green) [inline code/modules/surgery/organs/external/tails.dm:75]: This is why tails aren't working. I changed the formatting of the tails to m_tail_spikes_FRONT and m_waggingtail_spikes_FRONT to make it easier to work with. I also did this with functional wings IIRC, you'll need to rename your tails and wings in the .dmi to follow this formatting

**Review by Time-Green — COMMENTED** [review]
Impressed with the port! I'm afraid my .dmi fuckery is making stuff not render

File: code/modules/surgery/organs/external/tails.dm (hunk 1 above)
Comment (Jolly-66) [reply code/modules/surgery/organs/external/tails.dm:75]: Thank you for your help!

Comment (Time-Green) [reply code/modules/surgery/organs/external/tails.dm:75]: Okay so uhm I'm a fucking idiot and I wasnt on the right branch. The names are formatted as m_tail_lizard_smooth_BEHIND and m_tail_lizard_wagging_smooth_BEHIND

**Pull request merged**